### ISSUE TITLE
dont proxy e-ipfs traffic through cloudflare

### ISSUE THREAD

**Issue by olizilla** [web3-storage/ipfs-elastic-provider#31]
it's not totally clear if we should disable this, as cloudflare provides some comfortable tooling to see where traffic is coming from and block requesters... but if we did want to avoid the proxying here are the things i can see that need to change

- SSL cert provisioning - e-ipfs currently uses just the Cloudflare root cert. It would need to provision it's own.
- Firewall changes - allow direct connection to e-ipfs from any ip, not just cloudflare ranges
- DNS changes
  - create nameserver records for elastic.dag.hause on cloudflare and point all dns traffic for that to route53
  - create a records for the loadbalancer in route53
  - turn off proxying in cloudflare
- set up equivalent traffic monitoring and blocking tooling on aws as we have on cloudflare (??)

**Comment by olizilla** [comment]
@dchoi27 are you very very sure that you want this?

**Comment by dchoi27** [comment]
yes we do want this (though likely not without using cloudfront in front i'd assume given the blocking, monitoring, etc that we want - so think til we have answers forward there your evaluation of what this ticket would take is enough). what do you see as the big risks?

**Comment by dchoi27** [comment]
i do think that cloudfront doesn't have the same level of monitoring out of the box so think we'd have to set something up there. but i'd assume should be relatively sinple (just build on top of what we already have going in S3 and athena?)

**Comment by olizilla** [comment]
Risks
- time sink. a chunk of this work will have to happen in the terraform infrastructure project.
- monitoring and blocking tooling is an unknown, and i suspect will be less user friendly than what we have, and it increases the infra surface area we have to maintain. I don't have to do anything to keep the CF tools working.

**Comment by dchoi27** [comment]
yeah - i think if we end up reducing E-IPFS egress a ton from the denylist, and get better AWS egress costs via cloudfront, it might not be worth doing this depending on your spidey sense of how much risk of a time sink this is
